### PR TITLE
[MGDOBR-808] API Endpoint for Cloud Providers and Regions

### PR DESCRIPTION
**Pull request by robpblake**
[MGDOBR-808](https://issues.redhat.com/browse/MGDOBR-808)

This PR introduces three new endpoints on the RHOSE API:

- `/api/v1/cloud_providers`
- `/api/v1/cloud_providers/{id}`
- `/api/v1/cloud_providers/{id}/regions`

The first API endpoint lists the supported Cloud Providers into which a customer of RHOSE can deploy their Bridge instance. The second endpoint lists the regions within a specific Cloud Provider into which a customer of RHOSE can deploy their Bridge instance.

For the deployment of RHOSE for Service Preview, the only supported Cloud Provider is `AWS` with the `us-east-1` region. By default this configuration will be loaded from the classpath, however I've included the ability to override this configuration on a per-environment basis by mounting another instance of the `cloud_provider.json` at the location specified by the `rhose.config.external-directory` property (this defaults to `/config`). The override mechanism is generic and will feed into work being done by Vaibhav for https://issues.redhat.com/browse/MGDOBR-827

A follow-up PR to this will make it a requirement for the user to specify the `cloud_provider` and `region` when creating a Bridge instance.

## PR owner checklist

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config).  
  *Run `mvn clean verify -DskipTests` so that imports are correctly set.*
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] The layers in the `kustomize` folder have been updated accordingly.
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

## PR reviewer(s) checklist

- [ ] Check if code and Github action workflows have been modified and if the modification is safe
- [ ] If the modification is safe, add the `safe to test` label

<details>
<summary>
How to trigger pipelines and use the bots:
</summary>

- <b>Run the end to end pipeline</b>  
  Annotate the pull request with the label: `safe to test`. If you want to run the pipeline again, remove and add it again.

- <b>Rebase the pull request</b>  
  Comment with: `/rebase`.
- <b>Deploy BOT</b>
  Comment with `/deploy <dev,stable> when the PR has been merged to deploy to a target environment.

</details>

### CODE REVIEW

**Review by manstis — APPROVED** [review]
Comment about conformance to App Services API Guidelines but otherwise LGTM.

File: manager/src/test/resources/cloud_providers/cloud_providers.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    "regions": [
+      {
+        "id": "us-east-1",
+        "name": "us-east-1",
+        "display_name": "US East, N. Virginia",
+        "enabled": "true"
+      }
+    ]
+  }
+]
```
Comment (r00ta) [inline manager/src/test/resources/cloud_providers/cloud_providers.json:16]: why do you need the same `cloud_providers.json` in the test resources?

File: manager/src/main/java/com/redhat/service/smartevents/manager/api/user/CloudProviderAPI.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+    @Operation(summary = "List Supported Cloud Providers.", description = "Returns the list of supported Cloud Providers.")
+    @GET
+    public Response listCloudProviders(@Valid @BeanParam QueryPageInfo queryInfo) {
+        return Response.ok(ListResponse.fill(cloudProviderDAO.list(queryInfo), new CloudProviderListResponse(), CloudProviderResponse::from)).build();
+    }
+
+    @APIResponses(value = {
+            @APIResponse(description = "Success.", responseCode = "200",
+                    content = @Content(mediaType = MediaType.APPLICATION_JSON, schema = @Schema(implementation = CloudRegionListResponse.class))),
+            @APIResponse(description = "Bad request.", responseCode = "400", content = @Content(mediaType = MediaType.APPLICATION_JSON)),
```
Comment (sutaakar) [inline manager/src/main/java/com/redhat/service/smartevents/manager/api/user/CloudProviderAPI.java:55]:
```suggestion
            @APIResponse(description = "Bad request.", responseCode = "400", content = @Content(mediaType = MediaType.APPLICATION_JSON)),
            @APIResponse(description = "Not found.", responseCode = "404", content = @Content(mediaType = MediaType.APPLICATION_JSON)),
```

**Review by sutaakar — APPROVED** [review]

File: manager/src/main/java/com/redhat/service/smartevents/manager/api/models/responses/CloudProviderResponse.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package com.redhat.service.smartevents.manager.api.models.responses;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.redhat.service.smartevents.manager.models.CloudProvider;
+
+public class CloudProviderResponse {
```
[the remark this answers is not in the document]
Comment (robpblake) [reply manager/src/main/java/com/redhat/service/smartevents/manager/api/models/responses/CloudProviderResponse.java]: Thanks @manstis . Added this.

File: manager/src/test/resources/cloud_providers/cloud_providers.json (hunk 1 above)
Comment (robpblake) [reply manager/src/test/resources/cloud_providers/cloud_providers.json]: Left-over from testing. Removed.

File: manager/src/main/java/com/redhat/service/smartevents/manager/api/models/responses/CloudRegionResponse.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
+package com.redhat.service.smartevents.manager.api.models.responses;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.redhat.service.smartevents.manager.models.CloudRegion;
+
+public class CloudRegionResponse {
```
[the remark this answers is not in the document]
Comment (robpblake) [reply manager/src/main/java/com/redhat/service/smartevents/manager/api/models/responses/CloudRegionResponse.java:6]: @manstis I've not added it to `CloudRegion` as they are not entities in their own right. As per the service guideline docs:

```
The representation of objects that aren't resources, and don't have their own
identity, doesn't need to include the id, href and name attributes described above.
```

**Review by manstis — APPROVED** [review]

**Pull request merged**